.github/workflows/docker-image.yml

```diff
@@ -0,0 +1,28 @@
+name: Build and publish Docker image
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+  publish:
+    name: Build and publish Docker image
+    runs-on: ubuntu-latest
+    steps: 
+      - name: Log in to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      
+      - name: Check out repository code
+        uses: actions/checkout@v2
```

Comment:
checkout is at v3 now, better use the newest version

Reply:
Fixed, thanks!